app/forms/embargo_date_params_filter.rb

```diff
@@ -9,7 +9,7 @@ def call(schema, params)
       next unless dfn[:embargo_date]
 
       name = dfn[:name]
-      date_attributes[name] = deserialize(params, name) if dfn[:assign_if].call(params)
+      date_attributes[name] = params['release'] == 'embargo' ? deserialize(params, name) : nil
```

Comment:
This wipes out the embargo date for a work that used to have an embargo date and then flipped to immediate release.